clients/pkg/promtail/targets/docker/target.go

```diff
@@ -148,14 +148,38 @@ func extractTs(line string) (time.Time, string, error) {
 	return ts, pair[1], nil
 }
 
+// https://devmarkpro.com/working-big-files-golang
+func readLine(r *bufio.Reader) (string, error) {
+	var (
+		isPrefix = true
+		err      error
+		line, ln []byte
+	)
+
+	for isPrefix && err == nil {
+		line, isPrefix, err = r.ReadLine()
+		ln = append(ln, line...)
+	}
+
+	return string(ln), err
+}
```

Comment:
Do you mind putting this function into a utility package, e.g. `clients/pkg/promtail/common`? I think it may be used in various places.

Reply:
I'm fine either way on that actually, we can also wait until we need it somewhere else to figure out the best package for it to go into, or what the interface should look like, ie. there might be more duplication to reduce if we refactor our the section in `process` for processing a whole file?

Reply:
Yeah I'm fine either way, happy to move it if that's the desired approach...